### PR TITLE
chore(storage): remove abort operation

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Remove AbortOperation

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [x] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [x] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15557)

### CODE REVIEW

**Review by SkyFan2002 — APPROVED** [review]

File: src/query/storages/fuse/src/operations/common/processors/sink_commit.rs [hunk 1]
```diff
@@ -441,12 +447,6 @@ where F: SnapshotGenerator + Send + 'static
                         if let Some(files) = &self.copied_files {
                             metrics_inc_commit_copied_files(files.file_info.len() as u64);
                         }
-                        for segment in self.abort_operation.segments.iter() {
-                            self.ctx.add_segment_location((
-                                segment.to_string(),
-                                SegmentInfo::VERSION,
-                            ))?;
```
Comment (dantengsky) [inline src/query/storages/fuse/src/operations/common/processors/sink_commit.rs:448]: I am not sure if this is safe, seems agg index or something else depends on this?

Comment (zhyass) [reply src/query/storages/fuse/src/operations/common/processors/sink_commit.rs:448]: > I am not sure if this is safe, seems agg index or something else depends on this?

This operation is safe, we have added the add_segment_location operation where new segments are generated.
For example: https://github.com/datafuselabs/databend/pull/15557/files#diff-7c640d95f5c24d091ee74645f89c05486bee365406e05fe6168483b3a6432edaR234-R235

File: src/query/storages/fuse/src/operations/common/processors/transform_mutation_aggregator.rs [hunk 2]
```diff
@@ -216,6 +210,10 @@ impl TableMutationAggregator {
     pub async fn apply(&mut self) -> Result<CommitMeta> {
         let appended_segments = std::mem::take(&mut self.appended_segments);
         let appended_statistics = std::mem::take(&mut self.appended_statistics);
+
+        let mut new_segment_locs = Vec::new();
+        new_segment_locs.extend(appended_segments.clone());
```
Comment (dantengsky) [inline src/query/storages/fuse/src/operations/common/processors/transform_mutation_aggregator.rs:215]: Here, I'm puzzled:

Although it seems like `appended_segments` should be appended to `new_segment_locs`, before this PR, only new segments of `MutationKind` that are not `Insert` were collected into the `abort_operation` (hope I get it correctly).

Am I missing something here? 

should `append_segments` have been collected into the `abort_operation` in the previous versions?
or is it related to the removal of  `self.abort_operation.merge(abort_operation)` while processing mutation log entry of kind `MutationLogEntry::AppendSegment`?

Comment (zhyass) [reply src/query/storages/fuse/src/operations/common/processors/transform_mutation_aggregator.rs:215]: > should `append_segments` have been collected into the `abort_operation` in the previous versions? or is it related to the removal of `self.abort_operation.merge(abort_operation)` while processing mutation log entry of kind `MutationLogEntry::AppendSegment`?

It was previously collected in the `abort_operation` of `MutationLogEntry::AppendSegment`, the code is located https://github.com/datafuselabs/databend/blob/v1.2.470-nightly/src/query/storages/fuse/src/operations/common/processors/transform_serialize_segment.rs#L203-L207

And then merged into the `abort_operation` of `TransformSerializeSegment`

**Pull request merged**